src/Altinn.App.Core/Features/Validation/Default/ExpressionValidator.cs

```diff
@@ -92,7 +95,7 @@ IAppMetadata appMetadata
                     rowIndices: DataModel.GetRowIndices(resolvedField),
                     rowLength: null
                 );
-                var positionalArguments = new[] { resolvedField };
+                var positionalArguments = new object[] { resolvedField };
```

Comment:
In app-frontend, I added the dataTypeId as a second positional argument to be able to make generic validation definitions work with multiple data models.

https://github.com/Altinn/app-frontend-react/blob/092186ddde46021a89a8943761ffab8abda811b8/src/features/validation/expressionValidation/ExpressionValidation.tsx#L60

Then you can do this to refer to the "current" field being validated like this:
<img width="647" alt="image" src="https://github.com/Altinn/app-lib-dotnet/assets/47412359/c2b12e3f-a984-43e2-bd67-08a5048e3ac5">

Reply:
Not sure I fully understand, but is this what you look for?
```suggestion
                var positionalArguments = new object[] { resolvedField, baseField.DataType };
```

Reply:
How have you extended validation to support more data models? They are still in separate files per data model, which makes the data model in argv implicit from the filename?

Found a bug in the current implementation, because I don't traverse `*validation.json` to ensure that I prefetch all references to external models. This means that lookups in types not referenced in the layout will read `null` even if there is a value

Reply:
Yeah we could say that the data type should be implied by the file. I wasn't sure if we should do that or not though, since there would be a different "default" than in the layout files. But if we agree on that then I will change it in app-frontend as well.

Reply:
Isn't the premise for expression validation that it is different from layout, because it is tied to the data type and not the component?

Reply:
Absolutely, as long as we are on the same page between backend and frontend

Reply:
So, do we add `["argv", 1]` or not. I'm fine either way, but we need to do the same.

Reply:
I will make the default one follow the file, so I suppose it is not necessary to do both?

Reply:
I have removed argv 1 now